source/vir/src/prelude.rs

```diff
@@ -273,6 +280,63 @@ pub(crate) fn prelude_nodes(config: PreludeConfig) -> Vec<Node> {
         )))
         (axiom ([sized] [decorate_nil_sized]))
 
+        // The <T as Pointee>::Metadata projection type is special. Like Sized-ness, it depends
+        // only on the decoration.
+
+        // For Sized types: Metadata = ()
```

Comment:
Just one request: can you generate these axioms only when Pointee or PointeeMetadata is used, similar to how `prelude::array_functions` and `prelude::strslice_functions` work?